SwiftGit2 fails SwiftLint 0.17.0

Carthage failed to build SwiftGit2 with Xcode 8.2.1 with following errors:

```
/bin/sh -c /Users/groland/Library/Caches/org.carthage.CarthageKit/DerivedData/SwiftGit2/v0.3/Build/Intermediates/SwiftGit2.build/Release-iphoneos/SwiftLint.build/Script-C9CE0DD61E0710C20053205D.sh
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'CheckoutStrategy.swift' (1/17)
Linting 'Credentials.swift' (2/17)
Linting 'Errors.swift' (3/17)
/Users/groland/Desktop/gittest/Carthage/Checkouts/SwiftGit2/SwiftGit2/Credentials.swift:20:7: error: Identifier Name Violation: Enum element name should start with a lowercase character: 'Default' (identifier_name)
/Users/groland/Desktop/gittest/Carthage/Checkouts/SwiftGit2/SwiftGit2/Credentials.swift:21:7: error: Identifier Name Violation: Enum element name should start with a lowercase character: 'Plaintext' (identifier_name)
Linting 'Libgit2.swift' (4/17)
Linting 'Objects.swift' (5/17)
Linting 'OID.swift' (6/17)
Linting 'Pointers.swift' (7/17)
/Users/groland/Desktop/gittest/Carthage/Checkouts/SwiftGit2/SwiftGit2/Pointers.swift:26:7: error: Identifier Name Violation: Enum element name should start with a lowercase character: 'Commit' (identifier_name)
/Users/groland/Desktop/gittest/Carthage/Checkouts/SwiftGit2/SwiftGit2/Pointers.swift:27:7: error: Identifier Name Violation: Enum element name should start with a lowercase character: 'Tree' (identifier_name)
/Users/groland/Desktop/gittest/Carthage/Checkouts/SwiftGit2/SwiftGit2/Pointers.swift:28:7: error: Identifier Name Violation: Enum element name should start with a lowercase character: 'Blob' (identifier_name)
/Users/groland/Desktop/gittest/Carthage/Checkouts/SwiftGit2/SwiftGit2/Pointers.swift:29:7: error: Identifier Name Violation: Enum element name should start with a lowercase character: 'Tag' (identifier_name)
Linting 'References.swift' (8/17)
Linting 'Remotes.swift' (9/17)
Linting 'Repository.swift' (10/17)
/Users/groland/Desktop/gittest/Carthage/Checkouts/SwiftGit2/SwiftGit2/References.swift:133:7: error: Identifier Name Violation: Enum element name should start with a lowercase character: 'Lightweight' (identifier_name)
/Users/groland/Desktop/gittest/Carthage/Checkouts/SwiftGit2/SwiftGit2/References.swift:136:7: error: Identifier Name Violation: Enum element name should start with a lowercase character: 'Annotated' (identifier_name)
Linting 'Fixtures.swift' (11/17)
Linting 'FixturesSpec.swift' (12/17)
Linting 'ObjectsSpec.swift' (13/17)
/Users/groland/Desktop/gittest/Carthage/Checkouts/SwiftGit2/SwiftGit2/Repository.swift:18:73: warning: Redundant Void Return Violation: Returning Void in a function declaration is redundant. (redundant_void_return)
Linting 'OIDSpec.swift' (14/17)
Linting 'ReferencesSpec.swift' (15/17)
Linting 'RemotesSpec.swift' (16/17)
Linting 'RepositorySpec.swift' (17/17)
Done linting! Found 9 violations, 8 serious in 17 files.

** BUILD FAILED **

The following build commands failed:
        PhaseScriptExecution Lint\ Sources /Users/groland/Library/Caches/org.carthage.CarthageKit/DerivedData/SwiftGit2/v0.3/Build/Intermediates/SwiftGit2.build/Release-iphoneos/SwiftLint.build/Script-C9CE0DD61E0710C20053205D.sh
(1 failure)
```

Apparently enum identifiers with uppercase initials no longer conform to [recommended swift3 style](https://github.com/realm/SwiftLint/releases/tag/0.17.0).
I added ``identifier_name`` to ``disabled_rules`` in ``.swiftlint.yml`` as a temporary measure to circumvents the issue.

--- a/.swiftlint.yml
+++ b/.swiftlint.yml
@@ -3,6 +3,7 @@ disabled_rules:
   - force_cast
   - force_try
   - function_body_length
+  - identifier_name
   - type_body_length
 
 opt_in_rules:
@@ -12,7 +13,9 @@ opt_in_rules:
   - conditional_returns_on_newline
   - empty_count
   - explicit_init
+  - fatal_error_message
   - first_where
+  - implicitly_unwrapped_optional
   - overridden_super_call
   - private_outlet
   - prohibited_super_call

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,6 @@
 language: objective-c
 
-osx_image: xcode8.2
+osx_image: xcode8.3
 
 sudo: false # Enable container-based builds
 

--- a/SwiftGit2.xcodeproj/project.pbxproj
+++ b/SwiftGit2.xcodeproj/project.pbxproj
@@ -729,7 +729,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if which swiftlint >/dev/null; then\n    swiftlint\nelse\n    echo \"warning: SwiftLint is not installed\"\nfi";
+			shellScript = "if which swiftlint >/dev/null; then\n    swiftlint --lenient\nelse\n    echo \"warning: SwiftLint is not installed\"\nfi";
 		};
 /* End PBXShellScriptBuildPhase section */
 
@@ -1180,6 +1180,7 @@
 				C9CE0DD51E0710BD0053205D /* Release */,
 			);
 			defaultConfigurationIsVisible = 0;
+			defaultConfigurationName = Release;
 		};
 /* End XCConfigurationList section */
 	};

--- a/SwiftGit2/Repository.swift
+++ b/SwiftGit2/Repository.swift
@@ -15,7 +15,7 @@ public typealias CheckoutProgressBlock = (String?, Int, Int) -> Void
 /// Helper function used as the libgit2 progress callback in git_checkout_options.
 /// This is a function with a type signature of git_checkout_progress_cb.
 private func checkoutProgressCallback(path: UnsafePointer<Int8>?, completedSteps: Int, totalSteps: Int,
-                                      payload: UnsafeMutableRawPointer?) -> Void {
+                                      payload: UnsafeMutableRawPointer?) {
 	if let payload = payload {
 		let buffer = payload.assumingMemoryBound(to: CheckoutProgressBlock.self)
 		let block: CheckoutProgressBlock

